File: nvidia_chat_client.py
# Arquivo: nvidia_chat_client.py
#
# Terminal de Chat Interativo: Permite que o usuário converse com o modelo
# 'meta/llama3-8b-instruct' da NVIDIA, mantendo o histórico da conversa.
#
# Pré-requisito: Você deve ter a biblioteca 'openai' instalada:
# pip install openai
#
# Para executar: python nvidia_chat_client.py

from openai import OpenAI
import sys

# --- Configuração da API ---
NVIDIA_API_KEY = ""
NVIDIA_BASE_URL = "https://integrate.api.nvidia.com/v1"
LLM_MODEL = "meta/llama3-8b-instruct" # Modelo Conversacional comprovadamente funcional

# Inicializa o histórico de mensagens com a saudação
messages = [
    {"role": "assistant", "content": "Olá! Eu sou um assistente de IA. Qual é a sua primeira pergunta sobre tecnologia?"}
]

print("-" * 50)
print(f"Bem-vindo ao Chat Terminal (Modelo: {LLM_MODEL})")
print("Digite 'sair' ou 'exit' para encerrar a conversa.")
print("-" * 50)

# Imprime a primeira mensagem do assistente
print(f"Assistente: {messages[0]['content']}")


try:
    # 1. Configura o cliente da OpenAI para apontar para o endpoint da NVIDIA
    client = OpenAI(
      base_url = NVIDIA_BASE_URL,
      api_key = NVIDIA_API_KEY
    )

    while True:
        # 2. Obtém a entrada do usuário
        user_input = input("\nVocê: ")

        # Verifica condição de saída
        if user_input.lower() in ["sair", "exit"]:
            print("\nEncerrando a sessão. Até mais!")
            break

        # Adiciona a mensagem do usuário ao histórico
        messages.append({"role": "user", "content": user_input})

        # 3. Faz a chamada à API com o histórico completo
        print(f"Enviando solicitação à API da NVIDIA...")
        completion = client.chat.completions.create(
          model=LLM_MODEL,
          messages=messages, # Envia todo o histórico
          stream=False
        )

        # 4. Processa e imprime a resposta do assistente
        assistant_response = completion.choices[0].message.content
        messages.append({"role": "assistant", "content": assistant_response})
        
        print(f"Assistente: {assistant_response}")

except Exception as e:
    # Tratamento de erro detalhado
    if "401" in str(e):
        print(f"\n[ERRO] Ocorreu um erro 401 (Não Autorizado). Verifique a validade da sua chave de API.")
    elif "404" in str(e):
        print(f"\n[ERRO] Ocorreu um erro 404 (Modelo Não Encontrado).")
        print(f"Verifique se o modelo '{LLM_MODEL}' está ativado para sua chave no painel da NVIDIA.")
    else:
        print(f"\n[ERRO] Ocorreu um erro inesperado: {e}")
        print("Verifique sua conectividade e a validade da chave/modelo.")

File: nvidia_chat_client.js
/**
 * Arquivo: nvidia_chat_client.js
 *
 * Demonstra como se conectar à API da NVIDIA (compatível com OpenAI) usando Node.js.
 *
 * Configuração:
 * 1. npm install openai
 * 2. Salve este arquivo.
 * 3. Execute: node nvidia_chat_client.js
 */

// Usamos 'require' para maior compatibilidade com diferentes ambientes Node.js.
const OpenAI = require('openai');

// --- Configuração da API ---
// Sua chave de API da NVIDIA.
const NVIDIA_API_KEY = "";
const NVIDIA_BASE_URL = 'https://integrate.api.nvidia.com/v1';

const client = new OpenAI({
  apiKey: NVIDIA_API_KEY,
  baseURL: NVIDIA_BASE_URL,
});

// A conversa traduzida para o português.
const messages = [
    {"role":"user","content":"Esqueci como encerrar um processo no Linux, você pode me ajudar?"},
    {"role":"assistant","content":"Claro! Para encerrar um processo no Linux, você pode usar o comando kill, seguido pelo Process ID (PID) do processo que deseja encerrar."},
    {"role": "user", "content": "Ótimo, por favor, forneça um comando específico para encontrar o PID."}
];

async function runSafetyGuardTest() {
    // Modelo de Segurança: Retorna JSON de classificação.
    const LLM_MODEL = "nvidia/llama-3.1-nemotron-safety-guard-8b-v3";
    console.log(`\n--- Teste 1: Modelo Safety Guard (${LLM_MODEL}) ---`);
    console.log("Aguardando JSON de Classificação de Segurança...");
    
    try {
        const completion = await client.chat.completions.create({
            model: LLM_MODEL,
            messages: messages,
            stream: false,
        });

        console.log("\n[Resposta do Modelo Safety Guard - JSON de Classificação]");
        console.log("Role:", completion.choices[0]?.message?.role);
        // O modelo Safety Guard retorna um JSON com a classificação de segurança.
        console.log("Content:", completion.choices[0]?.message?.content); 
        
    } catch (error) {
        console.error(`\nErro ao chamar Safety Guard API: ${error.message}`);
    }
}

async function runConversationalTest() {
    // NOVA TENTATIVA: Usando o prefixo 'meta/' para o modelo Llama 3 8B Instruct, que é comum na API da NVIDIA.
    const LLM_MODEL = "meta/llama3-8b-instruct"; 
    console.log(`\n--- Teste 2: Modelo Conversacional (${LLM_MODEL}) ---`);
    console.log("Aguardando Resposta Conversacional...");

    try {
        const completion = await client.chat.completions.create({
            model: LLM_MODEL,
            messages: messages,
            stream: false,
        });

        console.log("\n[Resposta do Modelo Conversacional]");
        console.log("Role:", completion.choices[0]?.message?.role);
        console.log("Content:", completion.choices[0]?.message?.content);
        
    } catch (error) {
        console.error(`\nErro ao chamar Conversational API: ${error.message}`);
        console.log("Atenção: A NVIDIA usa nomes de modelos muito específicos. Se o erro 404 persistir, a verificação no seu painel da NVIDIA é obrigatória.");
    }
}

async function main() {
    await runSafetyGuardTest();
    await runConversationalTest();
}

main();